src/Symfony/Component/HttpFoundation/ProblemJsonResponse.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+use Symfony\Component\HttpFoundation\Exception\ProblemJsonResponseException;
+
+/**
+ * Represents a JSON response with a Problem Details object.
+ *
+ * @author Abdellah Ramadan <ramadanabdel24@gmail.com>
+ */
+class ProblemJsonResponse extends Response
+{
+    public function __construct(
+        int $status = 500,
+        string $type = 'about:blank',
+        ?string $title = null,
+        ?string $detail = null,
+        ?string $instance = null,
+        ?array $extensions = [],
+    ) {
+        parent::__construct();
+
+        $this->statusCode = $status;
+
+        if ($status < 400 || $status > 599) {
+            throw new ProblemJsonResponseException(\sprintf('The status code "%s" is not a valid HTTP Status Code error.', $this->statusCode));
```

Comment:
```suggestion
            throw new UnexpectedValueException(\sprintf('The status code must be in the 400-599 range, %d given.', $this->statusCode));
```
with `use` statement for HttpFoundation's `UnexpectedValueException`